week-1-ts-ecosystem/w01e03-object-oriented-structure/src/product/product.ts

```diff
@@ -0,0 +1,36 @@
+import {ProductId, ProductProperties} from "./models/product-properties.model";
+const cuid = require('cuid');
+
+export abstract class Product {
+    private readonly _id: ProductId;
+    private readonly _name: string;
+    protected _price: number;
+    private _amount: number;
+
+    protected constructor({name, amount, price}: ProductProperties) {
+        this._name = name;
+        this._amount = amount;
+        this._price = price ?? 0;
+        this._id = cuid();
+    }
+
+    get id(): ProductId {
+        return this._id;
+    }
+
+    get amount(): number {
+        return this._amount;
+    }
+
+    get price(): number {
+        return this._price;
+    }
+
+    increaseAmount(increment: number) {
+        this._amount +=increment;
```

Comment:
Tutaj jest błąd że nie ma logiki sprawdzania.
Zobacz, kto powiedział że `increment` będzie dodatnie :)

Skoro w `decreaseAmount` sprawdzasz @zajewskp czy nie wyjedziemy ponizej `0` to tutaj też powinieneś.
Jaką masz gwarancję że nie zrobię tak:
```
product.increaseAmount(-200);
```